### DISCUSSION TITLE
Schema For Certificate

### DISCUSSION THREAD

**Discussion by priyanka-TL** [Sunbird-Lern/Community#21] [Query]
@amit-tarento @reshmi-nair

For the project use case on Manage Learn side, we are planning to integrate Sunbird RC for certificate generate similar to the way its done in courses. Refere JIRA - https://project-sunbird.atlassian.net/browse/SB-28231

We have so far succeeded in generating a certificate based on our needs using Sunbird RC in local and
now we need some clarification on the how do we do it on the platform. If you can help us understand the below points:-

- How does QR code verification happen for courses? Which API of Sunbird RC is used?
- What is the env key for getting token to call Sunbird RC APIs on server?
- WHere do we upload new schema for the project certificate and credential templates? And how does it get passed on to Sunbird RC during deployment?

cc - @aks30 @vijiurs @kiranharidas187 @varsha-sl @Prateek-slokam @aishwaryashikshalokam

**Comment by rhwarrier** [comment]
@reshmi-nair  could you please respond ?

**Reply by reshmi-nair** [reply]
@priyanka-TL 

- How does QR code verification happen for courses? Which API of Sunbird RC is used? - There is no API which provide verification. Verification is happening in the client side by decrypting the credentials using public key. Please check with Sunbird-Ed team for more details.

- What is the env key for getting token to call Sunbird RC APIs on server? - Any valid user token can be used to call RC APIs. API key also need to be passed in header.

- Where do we upload new schema for the project certificate and credential templates? And how does it get passed on to Sunbird RC during deployment? - New schema and credential template has to be uploaded in devops repo. Make necessary changes in helm charts to copy schema to rc-core for during deploy and credential template, context files in to corresponding blob storage. Pls refer to https://github.com/project-sunbird/sunbird-devops/pull/3272/ for more details.
